packages/libs/react-components/src/components/Grid/GridContainer.tsx

```diff
@@ -0,0 +1,39 @@
+import { ICSSPropType } from '../../typings';
+import { convertStyles } from '../../utils';
+
+import { styled } from './../../styles';
+
+import React, { FC, ReactNode } from 'react';
+
+export interface IGridContainerProps {
+  children?: ReactNode;
+  gap?: ICSSPropType;
+  templateAreas?: ICSSPropType;
+  templateRows?: ICSSPropType;
+  templateColumns?: ICSSPropType;
+}
+
+const BasicContainer = styled('div', {
+  display: 'grid',
+  gridTemplateColumns: 'repeat(12, 1fr)',
+  gridGap: '$1',
+});
+
+const GridContainer: FC<IGridContainerProps> = ({
+  children,
+  gap,
+  templateAreas,
+  templateRows,
+  templateColumns,
+}) => {
+  const styles = convertStyles({
+    gridGap: gap ?? {},
```

Comment:
Is this necessary? My understanding is that gap can only be undefined or the value itself so would this work if it were just written as the following? `gridGap: gap,`

Reply:
hmm..thinking about it again. I added undefined to the Primitive type (used in the convertStyles function)
makes it much better. thanks

Reply:
not needed anymore